lib/indexer/migration_field_comparer.rb

```diff
@@ -0,0 +1,54 @@
+module Indexer
+  class MigrationFieldComparer
```

Comment:
Is there a clearer name for this? Hesitant to keep naming stuff "migration" because there's at least 3 meanings of that word and I think it will confuse people later on.

Reply:
renamed to GovukIndexFieldComparer and added a comment about reasons for existence and what it does.